src/main/java/io/fabric8/maven/docker/AbstractDockerMojo.java

```diff
@@ -240,11 +249,26 @@ private void storeInPluginContext(List<ImageConfiguration> resolvedImages, Strin
     private DockerAccess createDockerAccess(String minimalVersion) throws MojoExecutionException, MojoFailureException {
         DockerAccess access = null;
         if (isDockerAccessRequired()) {
-            String dockerUrl = EnvUtil.extractUrl(dockerHost);
+            boolean isDefaults = machine == null;
+            if (isDefaults) {
+                // create instance to resolve system defines
+                // docker.machine.name and docker.machine.autoCreate
+                machine = new MachineConfiguration();
+            }
+            isDefaults &= machine.resolveDefines(new PluginParameterExpressionEvaluator(session, execution), getLog());
```

Comment:
Thats true, for deeper configuration objects @Parameter is ignored. I used therefor plain field initializers and a constructor to feed the two properties from the outside when the configuration is created because of setting the property. Please look into AbstractDockerMojo how this is resolved.